### PR TITLE
Cleanup models when code editors are removed

### LINKED ISSUE #3458: Error: model already exists with URI
I'm seeing a lot of this on Sentry (~300 occurences in the last 24 hours): https://sentry.io/organizations/sourcegraph/issues/986971497/?project=1334031&query=is%3Aunresolved&statsPeriod=14d

This error was added in https://github.com/sourcegraph/sourcegraph/pull/3322 cc @sqs

### PR DESCRIPTION
**Pull request by lguychard**
Fixes #3458

@sqs adding you as reviewer here because this is code you recently added.

##### Code Hosts

- [x] GitHub

##### Browsers

- [x] Chrome

### CODE REVIEW

File: client/browser/src/libs/code_intelligence/code_intelligence.tsx [hunk 1]
```diff
@@ -484,9 +483,12 @@ export function handleCodeHost({
                 const editorId = extensionsController.services.editor.addEditor(editorData)
                 const codeViewState: CodeViewState = {
                     subscriptions: new Subscription(),
-                    editors: [editorId],
                     roots: [{ uri: toRootURI(fileInfo), inputRevision: fileInfo.rev || '' }],
                 }
+                codeViewState.subscriptions.add(() => {
+                    extensionsController.services.editor.removeEditor(editorId)
+                    extensionsController.services.model.removeModel(uri)
```
Comment (felixfbecker) [inline client/browser/src/libs/code_intelligence/code_intelligence.tsx]: Is there a 1:1:1 mapping between code views, editors and models?

Comment (lguychard) [reply client/browser/src/libs/code_intelligence/code_intelligence.tsx]: There may be several editors per code view (currently at most two, in the case of a diff view - that case is handled in this PR as well).

The model service currently assumes that there is a 1:1 mapping between editor and model, and throws if trying to add an URI that already exists.

That being said, you're right that there could be cases where several code views/editors reference the same model (for instance on PR discussion views, where several reviews may reference the same file diff, or if several snippets are inserted for the same file@rev on an issue). I'll rework this PR to take such cases into account.

Comment (lguychard) [reply client/browser/src/libs/code_intelligence/code_intelligence.tsx]: Added https://github.com/sourcegraph/sourcegraph/pull/3548/commits/f108084a00aa23df13afe86e6ba7e8f00b01b085 which handles this, PTAL.

**Comment by felixfbecker** [comment]
@sqs could you give this a review?

**Comment by felixfbecker** [comment]
Does this also fix https://github.com/sourcegraph/sourcegraph/issues/3549 or is that separate?

**Comment by felixfbecker** [comment]
@sqs PTAL

**Comment by lguychard** [comment]
@felixfbecker this does not fix #3549, it's separate.

**Pull request merged**